packages/contracts/contracts/plugin/IPluginSetup.sol

```diff
@@ -3,8 +3,27 @@
 pragma solidity 0.8.10;
 
 import {PermissionLib} from "../core/permission/PermissionLib.sol";
+import {IDAO} from "../core/IDAO.sol";
 
 interface IPluginSetup {
+    /// @param helpers The address array of helpers (contracts or EOAs) associated with the plugin after the install or update.
+    /// @param permissions The array of multi-targeted permission operations to be applied by the `PluginSetupProcessor` to the installing or updating DAO.
+    /// @param actions The array of follow up actions that will be executed through dao once the plugin is installed/updated/uninstalled.
+    struct PreparedDependency {
+        address[] helpers;
+        PermissionLib.ItemMultiTarget[] permissions;
+        IDAO.Action[] actions;
```

Comment:
I am not be up-to-date. Can this be just `actions` in the future? Do we need to keep them separate?
If not, is `permissions` sufficient and do we really need actions? After `initialize` has happened, the plugin should already be in an initialized state so that now follow-up actions should be required. 
We should try to reduce complexity for builders here as the feedback we received already mentioned that our setup is convoluted.

Reply:
I added `Actions` as it might be pretty interesting feature. I get it that it becomes a little bit complex ! in the very late future, they should become just Actions and permission object removed. for now, We could even do just Actions where permissions are inside but in order to do this, plugin setup would need easier construction for permissions.

We could do all that in the next PRs once this is merged.

Reply:
The feedback of the user was that writing a `PluginSetup` was  "convoluted". We should not add a second array of structs.

`Actions` are more flexible/desirable for sure, and it is better to give `EXECUTE_PERMISSION` instead of `ROOT_PERMISSION` to the PSP. 

I also agree with the DX issue 
> plugin setup would need easier construction for permissions
that you point out

With allowing the 3rd party dev to pass `Action[]`, I see also the problem of an increased attack surface and increased difficulty in reviewing plugin installation proposals.

Long story short, I would now just include one of the two, either the permissions array or the actions array, not both.

Using permissions is the more conservative/minimalistic approach.
Using including actions is more future-proof.

Reply:
Removed Actions.